crates/wasi-provider/src/lib.rs

```diff
@@ -141,33 +141,54 @@ impl Provider for WasiProvider {
         // know it is done
         // let fut = async {
         //     let handles = self.handles.read().await;
-        //     let containers = handles.get(pod_key(&pod));
+        //     let containers = handles.get(key_from_pod(&pod));
         // };
         Ok(Status {
             phase: Phase::Running,
             message: None,
         })
     }
-    fn logs(&self, _pod: KubePod) -> Result<Vec<String>, failure::Error> {
-        // TODO: We'll need to probably pass the runtime, output method, or a
-        // combined stream of stdout and stderr into our hashmap so we can
-        // expose them here
-        unimplemented!()
+    fn logs(
+        &self,
+        namespace: String,
+        pod: String,
+        container: String,
+    ) -> Result<Vec<String>, failure::Error> {
+        let mut handles = self.handles.write().unwrap();
```

Comment:
We need a mutable handle in order to read from the buffer below. If there is some way to do this without a write lock, please let me know

Reply:
If it advances a reader, it is a write operation. So this is likely correct.